iOS Examples/CBDSceneKit/Source/SceneKitViewController.m

```diff
@@ -0,0 +1,103 @@
+//
+//
+
+#import "SceneKitViewController.h"
+#include "CardboardSDK.h"
+
+#import <SceneKit/SceneKit.h>
+#import <SpriteKit/SpriteKit.h>
+
+@interface SceneKitViewController() <CBDStereoRendererDelegate>
+{
+    SCNScene *scene;
+    
+    SCNNode *cameraNode;
+    SCNNode *cameraControlNode;
+    
+    SCNRenderer *renderer;
+}
+
+@end
+
+@implementation SceneKitViewController
+
+- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
+{
+    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
+    if (!self)
+    {
+        return nil;
+    }
+    
+    self.stereoRendererDelegate = self;
+    
+    return self;
+}
+
+- (void)setupRendererWithView:(GLKView *)glView
+{
+    [EAGLContext setCurrentContext:glView.context];
+    
+    glClearColor(0.25f, 0.25f, 0.25f, 1.0f);
+    
+    scene = [SCNScene sceneNamed:@"scene.scnassets/scene.dae"];
+    
+    // cameraControlNode is the parent of the Camera node, we can use this to position/rotate the camera without affecting the head tracking
+    cameraControlNode = [scene.rootNode childNodeWithName:@"CameraControl" recursively:true];
+    // cameraNode is the node that actually controls the head tracking applied to the camera
+    cameraNode = [scene.rootNode childNodeWithName:@"Camera" recursively:true];
+    
+    // SpriteKitCube is a cube in the scene which we will apply the SKScene as a texture (nothing in the scene but that's irrelevant)
+    SCNNode *spriteKitCube = [scene.rootNode childNodeWithName:@"SpriteKitCube" recursively:true];
+    spriteKitCube.geometry.firstMaterial.diffuse.contents = [[SKScene alloc] initWithSize:CGSizeMake(100.0f, 100.0f)];
+    
+    renderer = [SCNRenderer rendererWithContext:glView.context options:nil];
+    renderer.scene = scene;
+    renderer.pointOfView = cameraNode;
+    
+    // Example of moving the CameraControlNode while not affecting the head tracking
+    // The position and rotation of this node is where the camera would be looking if it was straight ahead
+    // The head tracking rotation appleid to the cameraNode is applied after this so you get both effects
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 5 * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
+        SCNAction *cameraMoveAction = [SCNAction moveTo:SCNVector3Make(-4.5f, -4.5f, 0.0f) duration:10.0f];
+        cameraMoveAction.timingMode = SCNActionTimingModeEaseInEaseOut;
+        [cameraControlNode runAction:cameraMoveAction];
+    });
+}
+
+- (void)shutdownRendererWithView:(GLKView *)glView
+{
+}
+
+- (void)renderViewDidChangeSize:(CGSize)size
+{
+}
+
+- (void)prepareNewFrameWithHeadViewMatrix:(GLKMatrix4)headViewMatrix
+{
+    // NOTE: We do the glClear here because of SpriteKit being used as a texture
+    // If you move this to the start of drawEyeWithEye you will see the broken scissor
+    // that will clear the left side of the screen when the right eye is drawn
+    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
```

Comment:
This doesn't work correctly out of the box on an *iPhone 6* running *iOS 9.2.1*. There are artifacts near some sides of the screen, apparently because the whole screen is not being cleared (see screenshots below).

Doing this instead fixes the problem:

```
glDisable(GL_SCISSOR_TEST);
glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT); glEnable(GL_SCISSOR_TEST);
```

Do you think that's a reasonable solution?

![img_5294](https://cloud.githubusercontent.com/assets/444313/13747363/cc7c820a-e9ef-11e5-8468-92f44dbec1fe.PNG)
![img_5293](https://cloud.githubusercontent.com/assets/444313/13747362/cc6b4530-e9ef-11e5-9505-869ceef24096.PNG)